src/listen/ldap_sync/proto_ldap_sync_ldap.c

```diff
@@ -0,0 +1,1325 @@
+	{ FR_CONF_OFFSET("changes_only", FR_TYPE_BOOL, sync_config_t, changes_only), .dflt = "yes" },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static CONF_PARSER option_config[] = {
+	{ FR_CONF_OFFSET("idle", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.keepalive_idle), .dflt = "60" },
+
+	{ FR_CONF_OFFSET("probes", FR_TYPE_UINT32, proto_ldap_sync_ldap_t, handle_config.keepalive_probes), .dflt = "3" },
+
+	{ FR_CONF_OFFSET("interval", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.keepalive_interval), .dflt = "30" },
+
+	{ FR_CONF_OFFSET("dereference", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.dereference_str) },
+	/* allow server unlimited time for search (server-side limit) */
+	{ FR_CONF_OFFSET("srv_timelimit", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.srv_timelimit), .dflt = "60" },
+	/* timeout for results */
+	{ FR_CONF_OFFSET("res_timeout", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.res_timeout), .dflt = "20" },
+	/* timeout on network activity */
+	{ FR_CONF_OFFSET("net_timeout", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.net_timeout), .dflt = "10" },
+	/* timeout for reconnections */
+	{ FR_CONF_OFFSET("reconnection_delay", FR_TYPE_TIME_DELTA, proto_ldap_sync_ldap_t, handle_config.reconnection_delay), .dflt = "5" },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static CONF_PARSER tls_config[] = {
+	{ FR_CONF_OFFSET("ca_file", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_ca_file) },
+	{ FR_CONF_OFFSET("ca_path", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_ca_path) },
+	{ FR_CONF_OFFSET("certificate_file", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_certificate_file) },
+	{ FR_CONF_OFFSET("private_key_file", FR_TYPE_FILE_INPUT, proto_ldap_sync_ldap_t, handle_config.tls_private_key_file) },
+	{ FR_CONF_OFFSET("start_tls", FR_TYPE_BOOL, proto_ldap_sync_ldap_t, handle_config.start_tls), .dflt = "no" },
+	{ FR_CONF_OFFSET("require_cert", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.tls_require_cert_str) },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static const CONF_PARSER global_config[] = {
+	{ FR_CONF_OFFSET("random_file", FR_TYPE_FILE_EXISTS, proto_ldap_sync_ldap_t, tls_random_file) },
+	{ FR_CONF_OFFSET("ldap_debug", FR_TYPE_UINT32, proto_ldap_sync_ldap_t, ldap_debug), .dflt = "0x0000" },
+
+	CONF_PARSER_TERMINATOR
+};
+
+static CONF_PARSER const proto_ldap_sync_ldap_config[] = {
+	/*
+	 *	LDAP server definition
+	 */
+	{ FR_CONF_OFFSET("server", FR_TYPE_STRING | FR_TYPE_MULTI | FR_TYPE_REQUIRED, proto_ldap_sync_ldap_t, handle_config.server_str) },
+	{ FR_CONF_OFFSET("port", FR_TYPE_UINT16, proto_ldap_sync_ldap_t, handle_config.port) },
+	{ FR_CONF_OFFSET("identity", FR_TYPE_STRING, proto_ldap_sync_ldap_t, handle_config.admin_identity) },
+	{ FR_CONF_OFFSET("password", FR_TYPE_STRING | FR_TYPE_SECRET, proto_ldap_sync_ldap_t, handle_config.admin_password) },
```

Comment:
These could be moved into common code as a preprocessor macro taking the structure and the field within the structure for the `fr_ldap_config_t`